python/GangaDirac/Lib/Utilities/DiracUtilities.py

```diff
@@ -214,6 +214,34 @@ def _checkProxy( delay=60, renew = True, shouldRaise = True, force = False ):
             last_modified_time = time.time()
 
 
+def getAccessURLs(lfns):
+    """
+    This is a function to get a list of the accessURLs
+    for a provided list of lfns.
+    """
+    # Get all the replicas
+    reps = execute('getReplicas(%s)' %lfns)
+    # Get the SEs
+    SEs = []
+    for lf in reps['Value']['Successful']:
+      for thisSE in reps['Value']['Successful'][lf].keys():
+        if thisSE in SEs:
+          continue
+        else:
+          SEs.append(thisSE)
+    myURLs = []
+    # Loop over the possible SEs and get the URLs of the files stored there.
+    # Remove the successfully found ones from the list and move on to the next SE.
+    for SE in SEs:
+      thisSEFiles = execute('getAccessURL(%s, "%s")' % (lfns , SE))['Value']['Successful']
+      for lfn in thisSEFiles.keys():
+        myURLs.append(thisSEFiles[lfn])
+        lfns.remove(lfn)
```

Comment:
This will change the list that has been sent into the function which is almost certainly **not** what's expected. Either do a `copy` of the list or just add to a `done_lfn_list` and compare the two.